service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -226,6 +225,18 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskStarted(
 				return nil, serviceerrors.NewTaskAlreadyStarted("Workflow")
 			}
 
+			// Assuming a workflow is running on a sticky task queue by a workerA.
+			// After workerA is dead for more than 10s, matching will return StickyWorkerUnavailable error when history
+			// trying to push a new workflow task. When history sees that error, it will fall back to push the task to
+			// its original normal task queue without clear its stickiness to avoid an extra persistence write.
+			// We will clear the stickiness here when that task is delivered to another worker polling from normal queue.
+			// The stickiness info is used by frontend to decide if it should send down partial history or full history.
+			// Sending down partial history will cost the worker an extra fetch to server for the full history.
+			if mutableState.IsStickyTaskQueueEnabled() &&
+				mutableState.TaskQueue().GetName() != req.PollRequest.TaskQueue.GetName() {
```

Comment:
`req.PollRequest.TaskQueue.GetName()` here is the partitioned name? I guess it doesn't matter since sticky queues are never partitioned

Reply:
That is right. I will add a comment to clarify.